Services/Component/classes/class.ilPluginSlot.php

```diff
@@ -236,37 +228,28 @@ function getPluginsInformation()
 			{
 				// directories
 				if (@is_dir($pl_dir."/".$file) && substr($file, 0, 1) != "." &&
-					is_file($pl_dir."/".$file."/plugin.php"))
-				{
-					
-					
+					$this->checkPluginPhpFileAvailability($file)
+				) {
 					$plugin = array();
-					
+
 					$plugin = ilPlugin::lookupStoredData($this->getComponentType(),
 						$this->getComponentName(), $this->getSlotId(), $file);
-						
+
 					// create record in il_plugin table (if not existing)
-					ilPlugin::createPluginRecord($this->getComponentType(),
-						$this->getComponentName(), $this->getSlotId(), $file);
-					
+					if(count($plugin) == 0) {
+						ilPlugin::createPluginRecord($this->getComponentType(),
+							$this->getComponentName(), $this->getSlotId(), $file);
+					}
+
 					$pdata = $ilPluginAdmin->getAllData($this->getComponentType(),
 						$this->getComponentName(), $this->getSlotId(), $file);
 
-					$plugin["version"] = $pdata["version"];
-					$plugin["id"] = $pdata["id"];
-					$plugin["is_active"] = $pdata["is_active"];
-					$plugin["inactive_reason"] = $pdata["inactive_reason"];
-					$plugin["needs_update"] = $pdata["needs_update"];
-					$plugin["ilias_min_version"] = $pdata["ilias_min_version"];
-					$plugin["ilias_max_version"] = $pdata["ilias_max_version"];
-					$plugin["activation_possible"] = $pdata["activation_possible"];
-					$plugin["responsible"] = $pdata["responsible"];
-					$plugin["responsible_mail"] = $pdata["responsible_mail"];
-					
+					$plugin = array_merge($plugin, $pdata);
+
 					$plugin["name"] = $file;
-					$plugin["plugin_php_file_status"] = $this->checkPluginPhpFileAvailability($file);
+					$plugin["plugin_php_file_status"] = true;
```

Comment:
This is correct now, but if would prefer you still use `$this->checkPluginPhpFileAvailability($file)`  instead of `true` because the enclosing `if` might very well change in the future and turn this change into an incorrect one.